samples/bluetooth/peripheral_hids_keyboard/src/main.c

```diff
@@ -0,0 +1,629 @@
+#include <soc.h>
+#include <assert.h>
+
+#include <settings/settings.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+
+#include <gatt/bas.h>
+#include <bluetooth/services/hids.h>
+#include <bluetooth/services/dis.h>
+
+#define DEVICE_NAME     CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME_LEN (sizeof(DEVICE_NAME) - 1)
+
+#define BASE_USB_HID_SPEC_VERSION   0x0101
+
+#define OUTPUT_REPORT_MAX_LEN            1
+#define INPUT_REPORT_KEYS_MAX_LEN        8
+#define OUTPUT_REPORT_BIT_MASK_CAPS_LOCK 0x02
+#define INPUT_REP_KEYS_REF_ID            0
+#define OUTPUT_REP_REF_ID                0
+#define MODIFIER_KEY_POS                 0
+#define SHIFT_KEY_CODE                   0x02
+#define SCAN_CODE_POS                    2
+#define KEYS_MAX_LEN                    (INPUT_REPORT_KEYS_MAX_LEN - \
+					SCAN_CODE_POS)
+
+#define CAPS_LOCK_LED        LED0_GPIO_PIN
+#define SHIFT_BUTTON         SW1_GPIO_PIN
+#define SHIFT_BUTTON_PORT    1
+#define SHIFT_BUTTON_PRESSED 0
+
+/* HIDs queue size. */
+#define HIDS_QUEUE_SIZE 10
+
+/* HIDS instance. */
+HIDS_DEF(hids_obj,
+	 OUTPUT_REPORT_MAX_LEN,
+	 INPUT_REPORT_KEYS_MAX_LEN);
+
+/* Required security level */
+#ifdef CONFIG_BT_GATT_HIDS_SECURITY_LEVEL_LOW
+static const bt_security_t sec_level = BT_SECURITY_LOW;
+#elif CONFIG_BT_GATT_HIDS_SECURITY_LEVEL_MED
+static const bt_security_t sec_level = BT_SECURITY_MEDIUM;
+#elif CONFIG_BT_GATT_HIDS_SECURITY_LEVEL_HIGH
+static const bt_security_t sec_level = BT_SECURITY_HIGH;
+#elif CONFIG_BT_GATT_HIDS_SECURITY_LEVEL_FIPS
+static const bt_security_t sec_level = BT_SECURITY_FIPS;
+#else
+static const bt_security_t sec_level = BT_SECURITY_LOW;
+#endif
+
+struct key_pack {
+	void *private;
+	u8_t *elements;
+	u8_t len;
+};
+
+/* Keyboard queue. */
+K_MSGQ_DEFINE(hids_queue,
+	      sizeof(struct key_pack),
+	      HIDS_QUEUE_SIZE,
+	      4);
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_GAP_APPEARANCE,
+		      (CONFIG_BT_DEVICE_APPEARANCE >> 0) & 0xff,
+		      (CONFIG_BT_DEVICE_APPEARANCE >> 8) & 0xff),
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA_BYTES(BT_DATA_UUID16_ALL,
+		      0x12, 0x18,       /* HID Service */
+		      0x0f, 0x18),      /* Battery Service */
+
+};
+
+static const struct bt_data sd[] = {
+	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
+};
+
+static u8_t hello_world_str[] = {
+	0x0b,	/* Key h */
+	0x08,	/* Key e */
+	0x0f,	/* Key l */
+	0x0f,	/* Key l */
+	0x12,	/* Key o */
+	0x28,	/* Key Return */
+};
+
+static u8_t caps_on_key_scan_str[] = {
+	0x06,	/* Key C */
+	0x04,	/* Key a */
+	0x13,	/* Key p */
+	0x16,	/* Key s */
+	0x12,	/* Key o */
+	0x11,	/* Key n */
+};
+
+static u8_t caps_off_key_scan_str[] = {
+	0x06,	/* Key C */
+	0x04,	/* Key a */
+	0x13,	/* Key p */
+	0x16,	/* Key s */
+	0x12,	/* Key o */
+	0x09,	/* Key f */
+};
+
+static struct device		*led_dev;
```

Comment:
nitpick: variable names in declarations are not aligned in Zephyr.

Reply:
Fixed